volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,777 @@
+            vads.append((vad.get_start(), vad.get_size(), filepath))
+
+        return vads
+
+    @classmethod
+    def get_all_vads_with_file_paths(
+        cls,
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        symbol_table_name: str,
+    ) -> Generator[
+        Tuple[interfaces.objects.ObjectInterface, str, ranges_type],
+        None,
+        None,
+    ]:
+        """
+        Yields each set of vads for a process that have a file mapped, along with the process itself and its layer
+        """
+        procs = pslist.PsList.list_processes(
+            context=context,
+            layer_name=layer_name,
+            symbol_table=symbol_table_name,
+        )
+
+        for proc in procs:
+            try:
+                proc_layer_name = proc.add_process_layer()
+            except exceptions.InvalidAddressException:
+                continue
+
+            vads = PESymbols.get_proc_vads_with_file_paths(proc)
+
+            yield proc, proc_layer_name, vads
+
+    @staticmethod
+    def get_process_modules(
+        context: interfaces.context.ContextInterface,
+        layer_name: str,
+        symbol_table: str,
+        filter_modules: Optional[filter_modules_type],
+    ) -> Dict[str, List[Tuple[str, int, int]]]:
+        """
+        Walks the process list and each process' VAD to determine the base address and size of wanted modules
+        """
+        proc_modules: Dict[str, List[Tuple[str, int, int]]] = {}
+
+        if filter_modules:
+            # create a tuple of module names for use with `endswith`
+            filter_modules_check = tuple([key.lower() for key in filter_modules.keys()])
+        else:
+            filter_modules_check = None
+
+        for _, proc_layer_name, vads in PESymbols.get_all_vads_with_file_paths(
+            context, layer_name, symbol_table
+        ):
+            for vad_start, vad_size, filepath in vads:
+                filename = PESymbols.filename_for_path(filepath)
+
+                if filter_modules_check and not filename.endswith(filter_modules_check):
+                    continue
+
+                # track each module along with the process layer and range to find it
+                if filename not in proc_modules:
+                    proc_modules[filename] = []
+
+                proc_modules[filename].append((proc_layer_name, vad_start, vad_size))
+
+        return proc_modules
+
+    def _generator(self) -> Generator[Tuple[int, Tuple[str, str, int]], None, None]:
+        kernel = self.context.modules[self.config["kernel"]]
+
+        if self.config["symbol"]:
+            filter_module = {
+                self.config["module"].lower(): {
+                    PESymbols.wanted_names: [self.config["symbol"]]
+                }
+            }
+
+        elif self.config["address"]:
+            filter_module = {
+                self.config["module"].lower(): {
+                    PESymbols.wanted_addresses: [self.config["address"]]
```

Comment:
Should these not be `ListRequirements` then?

Reply:
Running the plugin is just to check one address or symbol at a time. As we evolve vol3, hopefully users will never have to run this plugin direct as plugins resolve symbols on their own.

Reply:
Sure, but... why are we forcing the users to do them one by one, rather than allow them to do lists?  It makes literally no difference to the plugin or how it's run in single item mode?  It just allows you to stick extra values afterwards to do several of them?  Seems like an obvious helpful move, but it's certainly not a show stopper, just a suggestion.  Feel free to mark this as resolved if you'd like...